user/serializers.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "password", "email", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {"password": {"write_only": True, "min_length": 5}}
+
+    def create(self, validated_data):
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        password = validated_data.pop("password", None)
```

Comment:
The `update` method removes the `password` from `validated_data` before calling `super().update()`. This is correct, but ensure that the `password` field is always provided when updating a user, as it is required for setting a new password. If the password is optional during updates, consider handling cases where it might be `None` more explicitly.